internal/provisioner/plain/pkg/git/git.go

```diff
@@ -0,0 +1,83 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+const (
+	defaultDirectory = "./manifests"
+	repositoryName   = "repo"
+)
+
+type checkoutCmd struct {
+	v1alpha1.GitSource
+}
+
+func CloneCommandFor(s v1alpha1.GitSource) (string, error) {
+	cmd := &checkoutCmd{GitSource: s}
+	err := cmd.Validate()
+	if err != nil {
+		return "", err
+	}
+	return cmd.String(), nil
+}
+
+func (c *checkoutCmd) String() string {
+	var checkoutCommand string
+	var repository = c.Repository
+	var directory = c.Directory
+	var branch = c.Ref.Branch
```

Comment:
Can ref be nil here?

Reply:
Ref is a non-pointer type so I think we are safe to reference it?

Reply:
I think since Ref is required on the API we can get away with this type of blind indexing, although I agree it looks a bit strange. We could alternatively update Ref to be a pointer type.